includes/framework/QCacheProviderRedis.class.php

```diff
@@ -0,0 +1,96 @@
+<?php
+	
+	/**
+	 * Cache provider based on Redis.
+	 *
+	 * This adapter needs the predis/predis library to be installed.
+	 * Please see http://github.com/nrk/predis
+	 */
+	class QCacheProviderRedis extends QAbstractCacheProvider {
+		/** @var Predis\Client */
+		protected $objPredisClient;
+
+		/**
+		 * Construct the Memcache based cache provider
+		 *
+		 * @param array $objOptionsArray array of server options. Each item in the array contains an associative
+		 *                               arrays with options for the server to add to memcache
+		 *
+		 * @throws QCallerException
+		 */
+		public function __construct($objOptionsArray) {
+			// There must be keys named 'parameters' and 'options' in the configuration
+			if(!array_key_exists('parameters', $objOptionsArray) || !array_key_exists('options', $objOptionsArray)) {
+				// Needed keys do not exist
+				throw new QCallerException('The configuration parameters for creating predis client in the configuration file are wrong. The config array must contain the "parameters" and "options" keys');
+			}
+
+			if(class_exists('Predis\Client')) {
+				// We have the predis client
+				$this->objPredisClient = new Predis\Client($objOptionsArray['parameters'], $objOptionsArray['options']);
+			} else {
+				throw new QCallerException('QCacheProviderRedis expects the Predis library to be installed. Please see http://github.com/nrk/predis for more.');
+			}
+		}
+
+		/**
+		 * Get the object that has the given key from the cache
+		 * @param string $strKey the key of the object in the cache
+		 * @return object
+		 */
+		public function Get($strKey) {
+			return unserialize($this->objPredisClient->get($strKey));
+		}
+
+		/**
+		 * Set the object into the cache with the given key
+		 *
+		 * @param string $strKey                the key to use for the object
+		 * @param string $objValue              the object to put in the cache
+		 * @param null   $intExpireAfterSeconds Number of seconds after which the key will be expired
+		 *
+		 * @return void
+		 */
+		public function Set($strKey, $objValue, $intExpireAfterSeconds = null) {
```

Comment:
For Redis, I have updated the config file and put in the `_REDIS_CACHE_PROVIDER_DEFAULT_TTL_` named variable which works for only the cache (you can see the usage in this file as well). This is useful for caches. 

As far as the usability of **per set** is concerned, I see (in my application, at least) that sometimes we need to create short-living objects (like a list of objects which should be cached and updated every 10 minutes) while there are some which need updating a lot less frequently. In such a case, one may want to fine-tune the duration for which the value has to be cached.

Think like this: There is a post which we present to the client(s). That post is seldom going to change - so that post should be cached for as long as possible (a day's TTL would be fine for that one). At the same time, we calculate *6 other posts related to this post*. We want this list to change every 5 minutes so that the user sees new recommendations every 5 minutes. This thing should have a shorter TTL. This parameter enables such behavior. 

This is the reason I added the option.